draft-pauly-quic-datagram.md

```diff
@@ -179,40 +173,76 @@ Datagram Data:
+and can store the contents in memory.
 
 DATAGRAM frames MUST be protected with either 0-RTT or 1-RTT keys.
 
-## Datagram Identifiers {#datagram-id}
-
-Since several applications relying on datagrams have the need to identify which
-application-level flow a given datagram is a part of, DATAGRAM frames carry a
-datagram identifier. Applications that do not have a need for the identifier
-can use the value zero on their DATAGRAM frames and use the DATAGRAM_ID bit
-to omit sending the identifier over the wire. If an application uses a mixture
-of DATAGRAM frames with and without the DATAGRAM_ID bit set, the frames without
-it are assumed to be part of the application-level flow with Datagram ID zero.
-
-## Flow Control and Acknowledgements
-
-Although the DATAGRAM frame is not retransmitted upon loss detection,
-it does contribute to the maximum data for the overall connection. Packets that
-contain only DATAGRAM frames do need to be acknowledged, but
-implementations SHOULD defer and batch acknowledgements since
-the timing of these acknowledgements is not used for loss recovery.
-
-The DATAGRAM frame does not provide any explicit flow control signaling
-apart from the connection-level flow control. DATAGRAM frames are flow
-controlled only when the maximum data for the connection is hit, at which point
-the BLOCKED frame is sent.
-
-In cases in which a DATAGRAM frame is blocked due to connection-level
-flow control or congestion control, an implementation MAY drop the frame
-without sending it.
+## Flow Identifiers {#flow-id}
+
+Flow identifiers represent bidirectional flows of datagrams within a single QUIC
+connection. These are effectively equivalent to UDP ports, that allow basic
+demultiplexing of application data. Whenever one side of a connection
+receives a frame with a Flow ID was was not previously known, it MAY represent
+this to the application as a new flow of datagrams.
+
+The primary role of the QUIC transport towards the flow identifier is to provide
+a standard mechanism for demultiplexing application data flows, which may be
+destined for different processing threads in the application, akin to UDP sockets.
+
+Beyond this, a sender SHOULD ensure that DATAGRAM frames within a single flow
+are transmitted in order relative to one another. If multiple DATAGRAM frames can
+packed into a single packet, the sender SHOULD group them by Flow ID to
+promote fate-sharing within a specific flow and improve the ability to process batches
+of datagram messages efficiently on the receiver.
+
+Applications that do not have a need for the Flow ID can use the value zero on
+their DATAGRAM frames and clear the FLOW_ID bit to omit sending the identifier
+over the wire. If an application uses a mixture of DATAGRAM frames with and
+without the FLOW_ID bit set, the frames without it are assumed to be part of the
+application-level flow with Flow ID zero.
+
+## Acknowledgements
```

Comment:
"Acknowledgements of DATAGRAM frames" ?
The first time I read this, I thought it was the wrong section  ;-)